### ISSUE TITLE
Make bundle naming consistent between development and release

### ISSUE THREAD

**Issue by ndarilek** [koute/cargo-web#38]
This will probably get easier once Webpack or some other tool is better integrated, but right now I'm manually building a static PWA page for loading my assets. One minor gotcha is that, in development mode, the loader is served up as /js/app.js, while the release build seems to create files named my_crate.{js,wasm}.

I understand that this is fairly easy to mitigate by moving some files, but I wonder if a better solution might be serving up the files at /my_crate.{js,wasm} in development mode too? Then I could use the regular static assets unmodified between development and production if I like, simply copying src/static into the target directory and publishing that to my server.

I'm willing to try taking this on, I just want to know if there's some gotcha related to serving up assets based on their target name rather than the generic /js/app.js. Thanks.

**Comment by koute** [comment]
Yes, I was planning on moving away from the `js/app.js` convention and start using crate names as filenames** by default. This will also pave a way for something like `cargo web deploy` which will automatically emit all of the files (both compilation artifacts and static files) somewhere where they'll be ready to be served statically.

** - A side note: we want to use something known (like a crate name) instead of simply using the artifact filenames as-is as those are **not** always what you'd expect, for example the Emscripten WASM target (and only that target!) generates the WASM file with every `-` replaced by the `_` in its filename. So one gotcha would be to replace the original `foo_bar.wasm` path in the `.js` file generated by Emscripten with `foo-bar.wasm` where `foo-bar` is the crate name. (Or better yet - tell Emscripten through a linker flag to generate the filename we want by default; I'm not sure if it's possible though.)

**Comment by koute** [comment]
This is partially done on `master`. I still need to fix the `.wasm` name though.
